File: DotNet/CesiumLanguageWriter/Advanced/ICesiumSphericalValuePropertyWriter.cs
using System.Collections.Generic;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// A writer that can write a value as a spherical.
    /// </summary>
    public interface ICesiumSphericalValuePropertyWriter : ICesiumInterpolatablePropertyWriter
    {
        /// <summary>
        /// Writes the value as a spherical value <c>[Clock, Cone, Magnitude]</c>. The clock angle is measured in the XY plane from the positive X axis toward the positive Y axis. The cone angle is the angle from the positive Z axis toward the negative Z axis.
        /// </summary>
        /// <param name="value">The value.</param>
        void WriteSpherical(Spherical value);

        /// <summary>
        /// Writes the value as a spherical value <c>[Clock, Cone, Magnitude]</c>. The clock angle is measured in the XY plane from the positive X axis toward the positive Y axis. The cone angle is the angle from the positive Z axis toward the negative Z axis.
        /// </summary>
        /// <param name="dates">The dates at which the value is specified.</param>
        /// <param name="values">The values corresponding to each date.</param>
        void WriteSpherical(IList<JulianDate> dates, IList<Spherical> values);

        /// <summary>
        /// Writes the value as a spherical value <c>[Clock, Cone, Magnitude]</c>. The clock angle is measured in the XY plane from the positive X axis toward the positive Y axis. The cone angle is the angle from the positive Z axis toward the negative Z axis.
        /// </summary>
        /// <param name="dates">The dates at which the value is specified.</param>
        /// <param name="values">The values corresponding to each date.</param>
        /// <param name="startIndex">The index of the first element to write.</param>
        /// <param name="length">The number of elements to write.</param>
        void WriteSpherical(IList<JulianDate> dates, IList<Spherical> values, int startIndex, int length);
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/ICesiumSphericalValuePropertyWriter.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import java.util.List;
import javax.annotation.Nonnull;

/**
 *  
 A writer that can write a value as a spherical.
 

 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public interface ICesiumSphericalValuePropertyWriter extends ICesiumInterpolatablePropertyWriter {
    /**
    *  
    Writes the value as a spherical value {@code [Clock, Cone, Magnitude]}. The clock angle is measured in the XY plane from the positive X axis toward the positive Y axis. The cone angle is the angle from the positive Z axis toward the negative Z axis.
    
    

    * @param value The value.
    */
    void writeSpherical(@Nonnull Spherical value);

    /**
    *  
    Writes the value as a spherical value {@code [Clock, Cone, Magnitude]}. The clock angle is measured in the XY plane from the positive X axis toward the positive Y axis. The cone angle is the angle from the positive Z axis toward the negative Z axis.
    
    
    

    * @param dates The dates at which the value is specified.
    * @param values The values corresponding to each date.
    */
    void writeSpherical(List<JulianDate> dates, List<Spherical> values);

    /**
    *  
    Writes the value as a spherical value {@code [Clock, Cone, Magnitude]}. The clock angle is measured in the XY plane from the positive X axis toward the positive Y axis. The cone angle is the angle from the positive Z axis toward the negative Z axis.
    
    
    
    
    

    * @param dates The dates at which the value is specified.
    * @param values The values corresponding to each date.
    * @param startIndex The index of the first element to write.
    * @param length The number of elements to write.
    */
    void writeSpherical(List<JulianDate> dates, List<Spherical> values, int startIndex, int length);
}